src/api_manager/fetch_metadata.cc

```diff
@@ -35,7 +35,7 @@ const char kMetadataServiceAccountToken[] =
 // Initial metadata fetch timeout (1s)
 const int kMetadataFetchTimeout = 1000;
 // Maximum number of retries to fetch metadata
-const int kMetadataFetchRetries = 5;
```

Comment:
I believe this retry is used for both fetching gce_metadata and token.   We need to separate them.
For gce_metadata, just retry once,  for token, we may have to retry 5 times